modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -155,6 +158,15 @@ sealed class IO<out A> : IOOf<A> {
     // Pure can be replaced by its value
     override fun <B> map(f: (A) -> B): IO<B> = Suspend { Pure(f(a)) }
 
+    // Pure can be replaced by its value
+    override fun <B> mapFilter(f: (A) -> Option<B>): IO<B> =
+      Suspend {
+        f(a).fold(
+          { RaiseError(IllegalArgumentException("IO execution should yield a valid result")) },
```

Comment:
The error message here and below could use a bit more context, else it'll be hard to point what the origin is.

Reply:
Error message change to: "IO execution yield invalid result after mapFilter"